### PR TITLE
Gantt and FileManager: Change the Getting Started section, Describe concepts

### PR DESCRIPTION
**Pull request by AnnaKondratova**

### CODE REVIEW

**Review by LanaDX — APPROVED** [review]

File: concepts/05 Widgets/File Manager/00 Getting Started with File Manager/20 Bind the File Manager Widget to a File System.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
+Create a [file provider](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/) that allows you to access and modify file systems.
+
+[note]
+In this section, the object file system is used as a quick way to bind the **FileManager** widget to data. Refer to the [Bind to File Systems](/Documentation/Guide/Widgets/FileManager/Bind_to_File_Systems/) section for more information on supported file systems.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/00 Getting Started with File Manager/20 Bind the File Manager Widget to a File System.md]:
```suggestion
In this section, the object file system is used to quickly to bind the **FileManager** widget to data. Refer to the [Bind to File Systems](/Documentation/Guide/Widgets/FileManager/Bind_to_File_Systems/) section for more information on supported file systems.
```

File: concepts/05 Widgets/File Manager/00 Getting Started with File Manager/20 Bind the File Manager Widget to a File System.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+To bind the **FileManager** widget to a hierarchical data structure, create an [Array](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Array/) file provider and assign the array of hierarchical JSON objects to the provider's [data](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Array/Configuration/#data) option. The Array file provider automatically binds data objects to the widget if the data objects have the default 'name', 'size', 'dateModified', etc. fields in its structure. For example:
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/00 Getting Started with File Manager/20 Bind the File Manager Widget to a File System.md]:
```suggestion
To bind the **FileManager** widget to a hierarchical data structure, create an [Array](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Array/) file provider and assign the array of hierarchical JSON objects to the provider's [data](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Array/Configuration/#data) option. The Array file provider automatically binds data objects to the widget if the data objects have the default 'name', 'size', 'dateModified', etc., fields in their structure. For example:
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/10 Object File System.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
+The **FileManager** widget provides the Array file provider to manage files and folders in an in-memory array of JSON objects.
+
+![FileManager - Hierarchical Data Structure](Content/images/doc/20_1/FileManager/Hierarchical-file-system.png)
+
+Create an Array file provider and assign the array of hierarchical JSON objects to the provider's [data](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Array/Configuration/#data) option to bind the **FileManager** widget to a hierarchical data structure.
+
+The provider automatically binds data objects to the widget if the data objects have default fields in its structure. For example:
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/10 Object File System.md]:
```suggestion
The provider automatically binds data objects to the widget if the data objects have default fields in their structure. For example:
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/10 Object File System.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        items: [
+            // ...
+            // Nested data objects with the same structure
+            // ...
+        ]
+    },
+    // ...
+    ];
+
+In the example below, the **FileManager** widget displays hierarchical data stored in an in-memory array that contains field with conventional names.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/10 Object File System.md]:
```suggestion
In the example below, the **FileManager** widget displays hierarchical data stored in an in-memory array that contains fields with conventional names.
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md [hunk 5]
```diff
@@ -0,0 +1,118 @@
+The **FileManager** widget provides the [Remote](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Remote/) file provider to access files and folders located on the server. 
+
+![](Content/images/doc/20_1/FileManager/remote-file-provider.png)
+
+Assign the [Remote](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Remote/) file provider to the [fileProvider](/Documentation/ApiReference/UI_Widgets/dxFileManager/Configuration/#fileProvider) option to connect the widget to a file system located on the server. The Remote file provider exposes APIs to get a file system's hierarchy and manage the items.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md]:
```suggestion
Assign the [Remote](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Remote/) file provider to the [fileProvider](/Documentation/ApiReference/UI_Widgets/dxFileManager/Configuration/#fileProvider) option to connect the widget to a file system located on the server. The Remote file provider exposes APIs to get the file system hierarchy and to manage items.
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+
+Set the [endpointUrl](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Remote/Configuration/#endpointUrl) option to the Url at which the widget can get the file system items.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md]: at which the widget can get the file system items????? what are you trying to say

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md [hunk 7]
```diff
@@ -0,0 +1,63 @@
+Use the **FileManager** widget's [Custom](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Custom/) file provider to implement custom APIs to access and manage file systems. This provider allows you to handle each file operation manually. Use the custom provider when it's necessary to connect the widget to an API service with custom request or response format.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md]:
```suggestion
Use the **FileManager** widget's [Custom](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Custom/) file provider to implement custom APIs to access and manage file systems. This provider allows you to handle each file operation manually. Use the custom provider when it's necessary to connect the widget to an API service with a custom request or response format.
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    <!--JavaScript-->
+    $("#file-manager").dxFileManager({
+        fileProvider: new DevExpress.fileProviders.Custom({
+            // your code
+        }),        
+    });
+
+---
+
+Use the [getItems](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Custom/Configuration/#getItems) option to get file system items. The **[fieldName]Expr** options specify the attribute names that stores a file system item's key, name, size, modification date and so on.
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md]:
```suggestion
Use the [getItems](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Custom/Configuration/#getItems) option to get file system items. The **[fieldName]Expr** options specify the attribute names that store file system item keys, names, sizes, modification dates, and etc.
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+        fileProvider: new DevExpress.fileProviders.Custom({
+            getItems: function (pathInfo) {
+                // your code
+            }         
+        }),        
+    });
+
+---
+
+Implement APIs to handle file operations (add, delete, rename, and so on).
```
Comment (arminal) [inline concepts/05 Widgets/File Manager/10 Bind to File Systems/30 Custom File System.md]:
```suggestion
Implement APIs to handle file operations (add, delete, rename, and etc).
```

File: concepts/05 Widgets/Gantt/00 Getting Started with Gantt/20 Bind the Gantt to Data.md [hunk 10]
```diff
@@ -1,7 +1,7 @@
...
 
 [note]
-The **tasks** data source is required, whereas the data sources that contain [resources](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/resources '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resources'), [dependencies](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/dependencies '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/dependencies') and [resource assignments](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/resourceAssignments '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resourceAssignments') are optional.
+The **tasks** data source is required, whereas the data sources that contain [resources](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resources/), [dependencies](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/dependencies/) and [resource assignments](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resourceAssignments/) are optional.
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/00 Getting Started with Gantt/20 Bind the Gantt to Data.md]:
```suggestion
The **tasks** data source is required, whereas the data sources that contain [resources](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resources/), [dependencies](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/dependencies/), and [resource assignments](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/resourceAssignments/) are optional.
```

File: concepts/05 Widgets/Gantt/00 Getting Started with Gantt/30 Add Columns to the Gantt.md [hunk 11]
```diff
@@ -1,4 +1,4 @@
-Use the [columns](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/columns.md '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/#columns') option to specify columns that the **Gantt** widget should display in the task list. The widget gets the columns from the [tasks](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/tasks '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/tasks') data source.
+Use the [columns](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/#columns/) option to specify columns that the **Gantt** widget should display in the task list. The widget gets the columns from the [tasks](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/tasks/) data source.
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/00 Getting Started with Gantt/30 Add Columns to the Gantt.md]:
```suggestion
Use the [columns](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/#columns/) option to specify columns that the **Gantt** widget should display in the task list. The widget gets columns from the [tasks](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/tasks/) data source.
```

File: concepts/05 Widgets/Gantt/00 Getting Started with Gantt/40 Enable Editing.md [hunk 12]
```diff
@@ -1,4 +1,4 @@
-Set the [enabled](/api-reference/10%20UI%20Widgets/dxGantt/1%20Configuration/editing/enabled.md '/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/editing/#enabled') option **true** to allow users to edit tasks in the **Gantt**. 
+Set the [enabled](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/editing/#enabled/) option **true** to allow users to edit tasks in the **Gantt**.
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/00 Getting Started with Gantt/40 Enable Editing.md]:
```suggestion
Set the [enabled](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/editing/#enabled/) option to **true** to allow users to edit tasks in the **Gantt**. 
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md [hunk 13]
```diff
@@ -0,0 +1,114 @@
...
+
+A large task can be divided into **subtasks** that allow you to track the progress. 
+
+![DevExtreme Gantt Chart - Subtasks](/images/Gantt/subtasks.png)
+
+A **milestone** is a task with the same start and endpoint. Milestones can be used as checkpoints for important points in a project.
+
+![DevExtreme Gantt Chart - Milestone](/images/Gantt/milestone.png)
+
+The **Gantt** widget gets data for tasks from a data source ([dataSource](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/tasks/#dataSource)). Task objects from the data source are automatically bound to the widget if these objects have the default 'title', 'start', 'end', etc. data fields in its structure. For example:
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md]:
```suggestion
The **Gantt** widget gets data for tasks from a data source ([dataSource](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/tasks/#dataSource)). Task objects from the data source are automatically bound to the widget if these objects have the default 'title', 'start', 'end', etc., data fields in their structure. For example:
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md [hunk 14]
```diff
@@ -0,0 +1,114 @@
...
+    {
+        'id': 1,
+        'parentId': 0,
+        'title': 'Software Development',
+        'start': new Date('2019-02-21T05:00:00.000Z'),
+        'end': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31       
+    }
+
+The following example illustrates how to bind the widget to tasks stored in the data source that contains data fields with conventional names:
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md]:
```suggestion
The following example illustrates how to bind the widget to tasks stored in the data source, which contains data fields with conventional names:
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md [hunk 15]
```diff
@@ -0,0 +1,114 @@
...
+        'parentId': 0,
+        'title': 'Software Development',
+        'start': new Date('2019-02-21T05:00:00.000Z'),
+        'end': new Date('2019-07-04T12:00:00.000Z'),
+        'progress': 31
+    },
+    // your code
+    ];
+
+If the data source's field names differ from the standard field names mentioned above, use the **[fieldName]Expr** properties to map task title, progress, start/end points, and so on.
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/10 Task.md]:
```suggestion
If the data source's field names differ from the standard field names mentioned above, use the **[fieldName]Expr** properties to map task title, progress, start/end points, etc.
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/20 Dependency.md [hunk 16]
```diff
@@ -0,0 +1,74 @@
...
+
+- Finish to Finish (FF) - The predecessor task's end point specifies the successor task's end point.
+
+    ![DevExtreme Gantt Chart - Finish-To-Finish Dependency](/images/Gantt/dependency-type-ff.png)
+
+- Start to Finish (SF) - The predecessor task's start point specifies the successor task's end point. 
+
+    ![DevExtreme Gantt Chart - Finish-To-Finish Dependency](/images/Gantt/dependency-type-sf.png)
+
+The **Gantt** widget gets data for dependencies from a data source ([dataSource](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/dependencies/#dataSource)). Dependency objects from the data source are automatically bound to the widget if these objects have the default 'id', 'type', etc. data fields in its structure. For example:
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/20 Dependency.md]:
```suggestion
The **Gantt** widget gets data for dependencies from a data source ([dataSource](/Documentation/ApiReference/UI_Widgets/dxGantt/Configuration/dependencies/#dataSource)). Dependency objects from the data source are automatically bound to the widget if these objects have the default 'id', 'type', etc., data fields in their structure. For example:
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/20 Dependency.md [hunk 17]
```diff
@@ -0,0 +1,74 @@
...
+    var dependencies = [{
+        'id': 0,
+        'predecessorId': 1,
+        'successorId': 2,
+        'type': 0
+    },
+    // your code
+    ];
+
+If the data source's field names differ from the standard field names mentioned above, use the **[fieldName]Expr** properties to map dependency's id, type, and so on.
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/20 Dependency.md]:
```suggestion
If the data source's field names differ from the standard field names mentioned above, use the **[fieldName]Expr** properties to map dependency's id, type, etc.
```

File: concepts/05 Widgets/Gantt/10 Gantt Elements/30 Resource.md [hunk 18]
```diff
@@ -0,0 +1,97 @@
...
+        'id': 1,
+        'text': 'Management'       
+    }
+    // Resource Assignment
+    {        
+        'id': 0,
+        'taskId': 3,
+        'resourceId': 1
+    }
+The following example illustrates how to bind the widget to resource and resource assignment objects stored in the data sources that contains data fields with conventional names:
```
Comment (arminal) [inline concepts/05 Widgets/Gantt/10 Gantt Elements/30 Resource.md]:
```suggestion
The following example illustrates how to bind the widget to resources and resource assignment objects stored in the data sources that contain data fields with conventional names:
```

File: concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md (hunk 6 above)
Comment (AnnaKondratova) [reply concepts/05 Widgets/File Manager/10 Bind to File Systems/20 Remote File System.md]: Updated to...

Set the [endpointUrl](/Documentation/ApiReference/UI_Widgets/dxFileManager/File_Providers/Remote/Configuration/#endpointUrl) option to the Url of an endpoint used to access and modify a file system.

**Pull request merged**